### PR TITLE
Light Manipulation Spellbook, and new relic effect "BONUS_DODGE"

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary

```SUMMARY: Mods "Adds new spell and light manipulation themed spellbook"```

#### Purpose of change

Noticed that Magiclysm already had a lot of light manipulation spells, but had no themed spellbook for them.

#### Describe the solution

The book "Of Light and Falsehoods" will cover all your current photomancy needs.

 This also adds a new spell that increases dodge attemptsthrough light manipulation, and two enchanted rings that share this effect

### CODE REVIEW

**Comment by KorGgenT** [comment]
Notes:
- add a spell scroll for that new spell
- BONUS_DODGE does not currently have an implementation, see 33837. it should be somewhat easy if you find the places that dodge gets calculated and call the enchantment function there.

**Comment by John-Candlebury** [comment]
Thanks KorGgenT!
I'll mark this as a WIP until I implement BONUS_DODGE.

**Comment by John-Candlebury** [comment]
Implemented BONUS_DODGE modifier. This should be ready for review once again.

**Comment by KorGgenT** [comment]
so with a brand new character, this is my dodge:
image
then i cast the spell and get this: 
image
i change my dodge skill to 9: 
image
and then i get this: 
image

what do you think? should this be the standard behavior for this artifact effect? that's pretty dang strong for +100%... and it doesn't appear to exactly be +100% anyway.

**Comment by KorGgenT** [comment]
I originally wanted to increase or decrease the number of blocks available, like martial arts do, but maybe the name of that value is a little misleading. it's still in the early stages so i can call that something else if you think increasing your dodging ability is more intuitive for this value

**Comment by John-Candlebury** [comment]
Yeah it was certainly too strong. Just changed the bonus to a flat +2 levels. Which is still rather strong.

I honestly think both effects do different things worth having. Increasing dodge attempts is mostly useful and more valuable for characters who fight in melee range, while increasing actual dodge skill is also usefull for characters that would rather nor get hit at all but that might occasionally find themselves in melee range (pure mages mostly).

A direct increase of dodge skill also better simulates Dnd Mage Armor spell too.

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 1]
```diff
@@ -76,6 +76,22 @@
     "charges_per_use": 1,
     "artifact_data": { "charge_type": "ARTC_TIME" }
   },
+  {
+    "copy-from": "mring_silver",
+    "type": "TOOL_ARMOR",
+    "id": "mring_dodge_bonus_1",
+    "name": "eel ring",
+    "description": "A thin silver band ring, depicting an eel coiled on itself.  Allows you to dodge an extra attack per turn.",
+    "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "BONUS_DODGE", "add": 2 } ] } ] }
```
Comment (KorGgenT) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]:
```suggestion
    "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "BONUS_DODGE", "add": 1 } ] } ] }
```

File: data/mods/Magiclysm/items/spellbooks.json [hunk 2]
```diff
@@ -148,6 +148,21 @@
     "color": "magenta",
     "use_action": { "type": "learn_spell", "spells": [ "seismic_stomp", "create_atomic_lamp", "ethereal_grasp" ] }
   },
+  {
+    "id": "light_manipulation_spellbook",
+    "type": "GENERIC",
+    "name": "Of Light and Falsehoods",
+    "description": "A small white book, it subtly amplifies the ambient light around it.",
+    "weight": 430,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/spellbooks.json]:
```suggestion
    "weight": "430 g",
```

File: data/mods/Magiclysm/items/spellbooks.json [hunk 3]
```diff
@@ -148,6 +148,21 @@
...
+    "volume": 3,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/spellbooks.json]:
```suggestion
    "volume": "750 ml",
```

**Pull request merged**